### PR TITLE
Add binding for configured event for all Rise Elements

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Add binding for configured event for all `RiseElements`

Each element registers to the event
Allow optional callback or registering to send
an event on configured

Disable once configured is received

Add 10 `funcs` overflow reset and log warning

Wrap `configure` with a promise for async execution
Allow component to add handlers

Ignore events that bubbled from child elements
Element source must match

## Motivation and Context
Fix for:
https://github.com/Rise-Vision/rise-data-weather/issues/70
https://github.com/Rise-Vision/rise-embedded-template/issues/17

General improvement for handling the `configured` event

## How Has This Been Tested?
Tested various scenarios in both Player and Sharedschedules. Updated unit tests.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

**Comment by alex-deaconu** [comment]
Will open separate PRs on the branch for unit tests and removing the logging.

@olegrise @ezequielc please review. Thanks!

File: src/rise-helpers.js [hunk 1]
```diff
@@ -183,30 +189,98 @@ RisePlayerConfiguration.Helpers = (() => {
...
+
+    if ( !binding ) {
+      binding = {
+        listener: _createConfiguredListener( component ),
+        funcs: [],
+        topic: topic
+      };
+      _componentBindings[ componentId ] = binding;
+
+      console.warn( `Component ${componentId} configured handler attached count: ${Object.keys( _componentBindings ).length}` );
```
Comment (olegrise) [inline src/rise-helpers.js:238]: @alex-deaconu are you planning to remove `console.warn`?

Comment (alex-deaconu) [reply src/rise-helpers.js:238]: @olegrise yes, see my note above. There is a second PR https://github.com/Rise-Vision/common-template/pull/121 for this. Thanks.

File: src/rise-helpers.js [hunk 2]
```diff
@@ -183,30 +189,98 @@ RisePlayerConfiguration.Helpers = (() => {
...
-        component.dispatchEvent( new Event( "rise-presentation-play" ));
+  function _createConfiguredListener( component ) {
+    return function(event) {
+      if (event.target !== component) {
+        return;
       }
-    });
 
-    component.dispatchEvent( new CustomEvent( "start" ));
+      Promise.resolve()
```
Comment (olegrise) [inline src/rise-helpers.js:198]: @alex-deaconu it's probably a good idea to add comment here explaining why using `Promise.resolve()` was necessary.

Comment (alex-deaconu) [reply src/rise-helpers.js:198]: @olegrise added in https://github.com/Rise-Vision/common-template/pull/121. Thanks.

**Comment by olegrise** [comment]
It worth mentioning just for the record that before this PR the sequence of the "start" and "rise-presentation-play" events was guaranteed i.e. the "start" was always sent first. After this change the sequence is not guaranteed - the events can come in any order.

File: src/rise-helpers.js (hunk 2 above)
Comment (olegrise) [reply src/rise-helpers.js:198]: @stulees I think you you need to know this. Adding `Promise.resolve()` fixes the issue https://github.com/Rise-Vision/rise-embedded-template/issues/17 by making the function execution asynchronous. 

The reason for issue 17 is that `rise-playlist` does not receive the `rise-presentation-play` event - the event is sent to the component before it registers an event handler. Here are the steps that lead to this situation:

1. common-template is loaded by template and configured
2. next `rise-playlist` is loaded. It `rise-playlist` receives `ready` event [here](https://github.com/Rise-Vision/rise-playlist-new/blob/master/src/rise-playlist.js#L183)
3. it starts processing `ready` event and the first line is `super.ready();` which takes it to the `rise-common-component` [here](https://github.com/Rise-Vision/rise-common-component/blob/master/src/rise-element.js#L69) 
4. the `RisePlayerConfiguration.isConfigured()` returns `true`, so the `_init` method is called right away and `configured` event is sent synchronously to the `common-template`
5. `common-template` receives `configured`
6. `common-template` calls `component.dispatchEvent( new Event( "rise-presentation-play" ));` which is also a synchronous call.
7. `rise-playlist` does not have an event handler for the `rise-presentation-play`, so it misses it
8. at this point the chain of synchronous calls triggered by `super.ready();` ends and execution proceeds to the next line [here](https://github.com/Rise-Vision/rise-playlist-new/blob/master/src/rise-playlist.js#L186) i.e. registering event handler for the `rise-presentation-play`, but it's too late.

The usage of `Promise.resolve()` makes the step 6 asynchronous allowing `rise-playlist` to finish processing `ready` event and registering all event handlers before `common-template` sends `rise-presentation-play` event.

Comment (stulees) [reply src/rise-helpers.js:199]: @olegrise Thanks for mentioning it and clarifying situation 👍

**Pull request merged**